### ISSUE TITLE
Question about the returned delta

### ISSUE THREAD

**Issue by YanghaoZYH** [ruoxi-jia-group/Universal_Pert_Cert#1]
Dear author,

Thanks for making the code public.

In the demo notebook, e.g. ```UP_Robust_Certification_Demo_MNIST.ipynb```, after solving the MILP problem, it seems that the returned universal delta can not lead to the desired cert_ACC.

```
# Validating the model ACC against UP
print((net(new_image+torch.tensor(delta).to(device).float().unsqueeze(0)).max(1)[1]==new_label.to(device)).sum().item()/len(new_image))
```
![image](https://github.com/ruoxi-jia-group/Universal_Pert_Cert/assets/22133105/2841f014-913f-4f38-b464-90f30126a079)

Are there any details I am missing or need to be noticed? 

Thanks in advance.

**Comment by shizhouxing** [comment]
Hi @YanghaoZYH , delta leads to the worst accuracy under *certification with linear bounds* by auto-lirpa (see Eq. (4) in the paper). The certified accuracy is a ceritifed lower bound of the actual accuracy you computed.

**Comment by YanghaoZYH** [comment]
I see, thanks for your quick response!
